### DISCUSSION TITLE
WYSIWYG images with captions

### DISCUSSION THREAD

**Discussion by matthanes** [directus/directus#20069] [Feature Requests]
### Summary

TinyMCE has the ability to add captions to images, but the "Add/Edit Image" button in Directus has a more custom interaction than the default TinyMCE behavior. There have been similar discussions, but I am including a potential solution which other discussions did not supply. I would propose adding to the Directus image interface a "Caption" field to like this:
![screenshot](https://github.com/directus/directus/assets/12828845/0e77ea36-0e5f-42b0-8adf-c60c1f88e825)

### Motivation

When a content editor is creating an article in a WYSIWYG field, they may want to give credit for an embedded image or provide context for the image using a caption. This is a different purpose than using alt text to describe the picture for screen readers.

### Detailed Design

As shown in the screenshot above, the interface could be given a new Caption field to supply the caption text. There appears to be a section of code starting [here](https://github.com/directus/directus/blob/9fdbae55f9f144843c4ab29fa241f6cf5eea11d5/app/src/interfaces/input-rich-text-html/useImage.ts#L159) that builds the `<img>` tag that is added back into the editor. 
![image](https://github.com/directus/directus/assets/12828845/8976b6f2-14e1-4262-9f60-ca2dfe6f22cf)

If possible, could there be an additional step that checks for the existence of a caption and if so, the `imageHtml` could be modified before being submitted to the editor like this:
```
<figure>
    ${imageHtml}
    <figcaption>${img.caption}</figcaption>
</figure>
```

### Must Have:
- Ability to insert a an image wrapped in a `<figure>` tag with a `<figcaption>` underneath.

### Drawbacks

This change may require adding some additional documentation. I would be happy to contribute to that change.

### Alternatives

I have tried to work at using custom formats and options overrides in the WYSIWYG interface but I could not grasp how to properly nest a figcaption inside a figure.

### Adoption Strategy

I cannot think of a reason this would not be backward compatible. If no caption text is given, then the behavior should remain the same to insert the default `<img>` tag.

**Comment by jeremyzilar** [comment]
@matthanes Thank you for submitting this 🎉 

Making a change to the markup that you propose would be a big improvement💯!

As for the caption field -- it would be nice if the fields that are already available in the `directus_files` collection are available within the TinyMCE Add Image panel. Currently it is only displaying a limited set of the fields. That way, if I add a `caption` field in addition to the `description` field, both would show up.

I should also note, as an editor of a publication, we often need separate fields for image rights, and image layout options. I would expect to be able to add these to the `directus_files` collection and have them available everywhere.

**Comment by HotpotCreative** [comment]
Bumping
A live edit for caption to travel with images would be great.
This option in Wordpress saves a lot of re-writing.

**Comment by thanhnndev** [comment]
Bumping x2 x3 x4

**Comment by matthanes** [comment]
Add your comments to this post while they are working on the new WYSIWYG editor: https://community.directus.com/t/new-tiptap-wysiwyg-editor/2439
